contrib/devtools/utxo_snapshot.sh

```diff
@@ -36,6 +36,16 @@ if (( GENERATE_AT_HEIGHT < PRUNED )); then
   exit 1
 fi
 
+# Check current block height to ensure the node has synchronized past the required block
+CURRENT_BLOCK_HEIGHT=$(${BITCOIN_CLI_CALL} getblockcount)
+PIVOT_BLOCK_HEIGHT=$(( GENERATE_AT_HEIGHT + 1 ))
```

Comment:
nit: Could reuse that below in the `PIVOT_BLOCKHASH=` line but only if you have to retouch.